test/extended/router/gatewayapicontroller.go

```diff
@@ -118,6 +158,10 @@ var _ = g.Describe("[sig-network-edge][OCPFeatureGate:GatewayAPIController][Feat
 		// check Subscription
 		waitVersionErr := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 20*time.Minute, false, func(context context.Context) (bool, error) {
 			csvName, err = oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "subscription", expectedSubscriptionName, "-o=jsonpath={.status.installedCSV}").Output()
+			if err != nil && strings.Contains(err.Error(), "not found") {
+				e2e.Logf("Subscription %q not found; retrying...", expectedSubscriptionName)
+				return false, nil
+			}
```

Comment:
I think that we should be consistent among all the polls we do in this block. I personally prefer how it's done for the OSSM deployment below:
```
		if err != nil {
				e2e.Logf("Failed to get OSSM operator deployment %q: %v; retrying...", deploymentOSSMName, err)
				return false, nil
			}
```
No assertions, just a retry for any error until a timeout. I think that some errors (not only "Not Found") can be temporary or intermittent.

Reply:
I was trying to keep my changes more narrowly focused.  All right, I can make the polling loop for the subscription retry on all errors.